### PR TITLE
fix flaky `lib/auth` cache-enabled tests

### PR DESCRIPTION
**Pull request by GavinFrazar**
Fixes flaky cache-enabled `lib/auth` tests, namely the `TestListUnifiedResources_*` tests (those are the only ones with cache enabled).

I was debugging an unrelated change locally and noticed sporadic failures in these test cases, for example:
```
=== Failed
=== FAIL: lib/auth TestListUnifiedResources_WithSearch (0.03s)
    tls_test.go:4610: 
                Error Trace:    /Users/gavin/work/teleport/lib/auth/tls_test.go:461
0
                                                        /Users/gavin/work/teleport/
lib/auth/auth_with_roles_test.go:4451
                Error:          Received unexpected error:
                                cluster name not found
                Test:           TestListUnifiedResources_WithSearch
```

The fix is to start the test auth server's cache after config resources like cluster name are all initialized, the same as we do in lib/service/service.go:initAuthService for the real implementation.

We only enabled cache in these tests since v14 in [this PR](https://github.com/gravitational/teleport/pull/33939), so that's as far as I'll backport.

### CODE REVIEW

File: lib/auth/helpers.go [hunk 1]
```diff
@@ -436,6 +437,13 @@ func NewTestAuthServer(cfg TestAuthServerConfig) (*TestAuthServer, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	// Auth initialization is done (including creation/updating of all singleton
+	// configuration resources) so now we can start the cache.
+	if c, ok := srv.AuthServer.Cache.(*cache.Cache); ok {
+		if err := c.Start(); err != nil {
```
Comment (zmb3) [inline lib/auth/helpers.go:443]: Is there a corresponding Stop that should happen on shutdown?

Comment (GavinFrazar) [reply lib/auth/helpers.go:443]: Hm, good point.

Looks like in the real impl the cache exits when the process exit context is done:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/service/service.go#L2214

But in tests we are using context.Background:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/auth/helpers.go#L235

Prior behavior was calling Start() as well:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/cache/cache.go#L918-L925

I can't find anywhere that actually calls `authServer.Cache.Close()`

Comment (GavinFrazar) [reply lib/auth/helpers.go:443]: ah, it's called internally when the context is done:
https://github.com/gravitational/teleport/blob/529bdbba0bf138b1a10124c07c70078fe5a48c19/lib/cache/cache.go#L1060-L1064

Comment (GavinFrazar) [reply lib/auth/helpers.go:443]: I'll update the tests to clean that up properly

Comment (GavinFrazar) [reply lib/auth/helpers.go:443]: fixed: [fd89faa](https://github.com/gravitational/teleport/pull/36216/commits/fd89faa8240b9c67440654628a9225039c486273)

**Review by fspmarshall — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

File: lib/auth/helpers.go (hunk 1 above)
Comment (zmb3) [reply lib/auth/helpers.go:443]: Nice detective work - thanks for the fix.

**Pull request merged**